test/python/transpiler/test_clifford_t_passmanager.py

```diff
@@ -0,0 +1,233 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2025.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test Clifford+T transpilation pipeline"""
+
+import numpy as np
+from ddt import ddt, data
+
+from qiskit.converters import circuit_to_dag
+from qiskit.circuit import QuantumCircuit
+from qiskit.circuit.library import QFTGate, iqp, GraphStateGate
+from qiskit.transpiler.passes.utils import CheckGateDirection
+from qiskit.transpiler.preset_passmanagers import generate_preset_pass_manager
+from qiskit.transpiler import CouplingMap
+from qiskit.providers.fake_provider import GenericBackendV2
+from qiskit.quantum_info import get_clifford_gate_names
+
+from test import QiskitTestCase  # pylint: disable=wrong-import-order
+
+
+@ddt
+class TestCliffordTPassManager(QiskitTestCase):
+    """Test Clifford+T transpilation pipeline."""
+
+    @data(0, 1, 2, 3)
+    def test_cliffords_1q(self, optimization_level):
+        """Clifford+T transpilation of a circuit with single-qubit Clifford gates."""
+        qc = QuantumCircuit(3)
+        qc.h(0)
+        qc.s(1)
+        qc.h(2)
+        qc.sdg(2)
+        qc.h(2)
+        qc.h(2)
+
+        basis_gates = ["cx", "s", "sdg", "h", "t", "tdg"]
+        pm = generate_preset_pass_manager(
+            basis_gates=basis_gates, optimization_level=optimization_level
+        )
+        transpiled = pm.run(qc)
+        self.assertLessEqual(set(transpiled.count_ops()), set(basis_gates))
+
```

Comment:
in this case, one should use only clifford gates for the transpilation?

Reply:
That would be true with the Clifford plugin I mentioned somewhere, but not yet. For efficiency, we are currently running the Solovay-Kitaev synthesis algorithm with the basis set ``["h", "t", "tdg"]``, so even pure Clifford gates would be transpiled into this basis. Note that you could run the SK algorithm with a much larger set of basis gates, like all Clifford gates + T/Tdg, however at least the python implementation becomes significantly slower. This is something to revisit when the SK algorithm is ported to Rust.